features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2

```diff
@@ -3121,6 +3121,14 @@
     </AllowedUserstores>
     {% endif %}
 
+    {% if user_store_mgt.reserved_user_stores is defined %}
+    <ReservedUserstores>
+        {% for reserved_user_store in user_store_mgt.reserved_user_stores %}
+        <ReservedUserstore>{{reserved_user_store}}</ReservedUserstore>
```

Comment:
IMO, we should define the default set of restricted domains in default.json file

Reply:
addressed with [44e1fb9](https://github.com/wso2/carbon-identity-framework/pull/4609/commits/44e1fb9c7335896977949dd3b8303a75da3f6245)